File: frontend/src/config/planLimits.ts
export type UiFirmPlan = "free" | "team" | "firm";

export type UiPlanLimits = {
  maxReviewsPerUpload: number | null;
  maxReportsPerMonth: number | null;
  maxUsers: number | null;
  pdfWatermark: boolean;
  historyDays?: number | null;
};

export function normalizeFirmPlan(plan?: string | null): UiFirmPlan {
  const value = (plan || "").toLowerCase();
  if (value === "team" || value === "professional") return "team";
  if (value === "firm" || value === "leadership") return "firm";
  return "free";
}

export function resolvePlanLimits(currentPlan?: {
  firmPlan?: string | null;
  planLimits?: Partial<UiPlanLimits> | null;
} | null): UiPlanLimits {
  const serverLimits = currentPlan?.planLimits;
  return {
    maxReviewsPerUpload:
      typeof serverLimits?.maxReviewsPerUpload === "number" ? serverLimits.maxReviewsPerUpload : null,
    maxReportsPerMonth:
      typeof serverLimits?.maxReportsPerMonth === "number" ? serverLimits.maxReportsPerMonth : null,
    maxUsers: typeof serverLimits?.maxUsers === "number" ? serverLimits.maxUsers : null,
    pdfWatermark: Boolean(serverLimits?.pdfWatermark),
    historyDays: typeof serverLimits?.historyDays === "number" ? serverLimits.historyDays : null,
  };
}

File: backend/services/plan_limits.py
"""Central, server-authoritative plan limits for Clarion."""

PLAN_LIMITS = {
    "free": {
        "max_users": 1,
        "max_reviews_per_upload": 50,
        "max_reports_per_month": 1,
        "history_days": 90,
        "pdf_watermark": True,
    },
    "team": {
        "max_users": 5,
        "max_reviews_per_upload": 5000,
        "max_reports_per_month": 10,
        "history_days": 365,
        "pdf_watermark": False,
    },
    "firm": {
        "max_users": 20,
        "max_reviews_per_upload": 5000,
        "max_reports_per_month": None,
        "history_days": None,
        "pdf_watermark": False,
    },
}

